be/src/exec/schema_scanner/schema_pipe_files.cpp

```diff
@@ -26,47 +27,86 @@ SchemaScanner::ColumnDesc SchemaTablePipeFiles::_s_columns[] = {
+        {"LAST_MODIFIED", TYPE_VARCHAR, sizeof(StringValue), false},
+
         {"LOAD_STATE", TYPE_VARCHAR, sizeof(StringValue), false},
+        {"STAGED_TIME", TYPE_VARCHAR, sizeof(StringValue), false},
+        {"START_LOAD_TIME", TYPE_VARCHAR, sizeof(StringValue), false},
+        {"FINISH_LOAD_TIME", TYPE_VARCHAR, sizeof(StringValue), false},
+
         {"ERROR_MSG", TYPE_VARCHAR, sizeof(StringValue), false},
-        {"ERROR_LINE_NUMBER", TYPE_BIGINT, sizeof(int64_t), false},
-        {"ERROR_COLUMN", TYPE_VARCHAR, sizeof(StringValue), false},
+        {"ERROR_COUNT", TYPE_BIGINT, sizeof(int64_t), false},
+        {"ERROR_LINE", TYPE_BIGINT, sizeof(int64_t), false},
 };
 
 SchemaTablePipeFiles::SchemaTablePipeFiles()
-    :SchemaScanner(_s_columns, sizeof(_s_columns) / sizeof(SchemaScanner::ColumnDesc)) ){
-}
+        : SchemaScanner(_s_columns, sizeof(_s_columns) / sizeof(SchemaScanner::ColumnDesc)) {}
 
 Status SchemaTablePipeFiles::start(RuntimeState* state) {
-    RETURN_IF_ERROR(SchemaScanner::start(state));
+    return SchemaScanner::start(state);
+}
 
-    return {};
+Status SchemaTablePipeFiles::_list_pipe_files() {
+    RETURN_IF(_param->ip == nullptr || _param->port == 0, Status::InternalError("unknown frontend address"));
+
+    TListPipeFilesParams params;
+    if (_param->current_user_ident) {
+        params.__set_user_ident(*_param->current_user_ident);
+    }
+    return SchemaHelper::list_pipe_files(*(_param->ip), _param->port, params, &_pipe_files_result);
 }
 
 Status SchemaTablePipeFiles::get_next(ChunkPtr* chunk, bool* eos) {
-    if (_cur_idx > 0) {
-        *eos = true;
-        return {};
+    while (_cur_row >= _pipe_files_result.pipe_files.size()) {
+        if (!_fetched) {
+            // send RPC
+            _fetched = true;
+            RETURN_IF_ERROR(_list_pipe_files());
+        } else {
+            *eos = true;
+            return Status::OK();
+        }
     }
-    fill_chunk(chunk);
     *eos = false;
-    return {};
+    return _fill_chunk(chunk);
+}
+
+DatumArray SchemaTablePipeFiles::_build_row() {
+    auto& pipe_file = _pipe_files_result.pipe_files.at(_cur_row);
+    _cur_row++;
+    return {
+            Slice(pipe_file.database_name),
+            pipe_file.pipe_id,
+            Slice(pipe_file.pipe_name),
+
+            Slice(pipe_file.file_name),
+            Slice(pipe_file.file_version),
+            pipe_file.file_rows,
+            pipe_file.file_size,
+            Slice(pipe_file.last_modified),
+
+            Slice(pipe_file.state),
+            Slice(pipe_file.staged_time),
+            Slice(pipe_file.start_load),
+            Slice(pipe_file.finish_load),
+
+            Slice(pipe_file.first_error_msg),
+            (int64_t)pipe_file.error_count,
+            (int64_t)pipe_file.error_line,
+    };
 }
 
-Status SchemaTablePipeFiles::fill_chunk(ChunkPtr* chunk) {
-    _cur_idx++;
+Status SchemaTablePipeFiles::_fill_chunk(ChunkPtr* chunk) {
```

Comment:
the result chunk contains only one row?

Reply:
yes, `get_next` only return one row each time